File: jscomp/test/rethrow_stacktrace_test.ml
(**
function g() { error; }
function f() { g(); }
function rethrow(e) { throw e; }
function test() {
  try {
    f();
  } catch (e) {
    rethrow(e);
  }
}
try { test() } catch (e) { console.log(e.stack) }
*)


let g () = 
  ignore [%raw {|"no inlining"|} ];
  raise Not_found  

let f () = 

  ignore [%raw {|"no inlining"|} ];
  g ()  

let rethrow e = raise e 


let test () = 
  try 
    f ()
  with 
    e -> raise e 
    
let () = 
    try test ()    
    with exn ->
      Js.log ((Obj.magic exn)##stack) 

File: jscomp/test/rethrow_stacktrace_test.js
'use strict';

var Caml_exceptions = require("../../lib/js/caml_exceptions.js");
var Caml_js_exceptions = require("../../lib/js/caml_js_exceptions.js");
var Caml_builtin_exceptions = require("../../lib/js/caml_builtin_exceptions.js");

function g(param) {
  (("no inlining"));
  throw Caml_exceptions.stacktrace(Caml_builtin_exceptions.not_found);
}

function f(param) {
  (("no inlining"));
  return g(/* () */0);
}

function rethrow(e) {
  throw Caml_exceptions.stacktrace(e);
}

function test(param) {
  try {
    return f(/* () */0);
  }
  catch (e){
    throw Caml_exceptions.stacktrace(e);
  }
}

try {
  test(/* () */0);
}
catch (raw_exn){
  var exn = Caml_js_exceptions.internalToOCamlException(raw_exn);
  console.log(exn.stack);
}

exports.g = g;
exports.f = f;
exports.rethrow = rethrow;
exports.test = test;
/*  Not a pure module */
